contracts/core/Controller.sol

```diff
@@ -718,7 +698,7 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
 
         // store new vault
         accountVaultCounter[_args.owner] = vaultId;
-        vaultType[_args.owner][vaultId] = _args.vaultType;
+        vaultType[_args.owner][vaultId] = borrowablePool.isWhitelistedOptionsVault(_args.owner) ? 2 : _args.vaultType;
```

Comment:
I think here better to keep it as the original implementation, and call `borrowablePool.isWhitelistedOptionsVault()` to check if whitelisted only if `_args.vaultType == 2`, it is more flexible as it allow vault owner even when whitelisted to open naked margin/spread vault, and also make the external call to check if whitelisted only if type==2

Reply:
The `_args.owner` that are whitelisted are exclusively the Ribbon DOV vault addresses themselves which generate the vault. We don't plan on having them be naked margin but if we do we will only be constraining ourselves which is fine. As for spread vault, referring to my comments [here](https://github.com/opynfinance/GammaProtocol/pull/459#discussion_r929257240) shouldn't MarginCalculator treat vault type 2 same as vault type 1 (spread/max loss vault).

Reply:
Yes I agree, but in more generic way for Gamma, it is better to keep it more flexible for other use cases, right? as this assume `_args.owner` will always only be a ribbon DOV

Reply:
I'm not sure I get `Also, making it _args.vaultType == 2 requires us to change the code for and redeploy all the vaults that plan to be using this.` ?

A solution like the below allow `owner` to open a spread vault, naked margin and borrowable one(assuming owner is whitelisted)

```
if (_args.vaultType == 2) {
  require(borrowablePool.isWhitelistedOptionsVault(_args.owner));
}
vaultType[_args.owner][vaultId] = _args.vaultType;
```

Reply:
Do u meant if implementing my suggestion, you will need to redeploy RIbbon vault? 
If so, how where you planning to handle that with the initial implementation https://github.com/opynfinance/GammaProtocol/pull/459#discussion_r924390199 ?

Reply:
Yes in general I agree your approach is better design (letting owner pass it through) instead of overriding whatever value is passed. Currently we pass in vault type 0 in these vaults which will be overridden with this upgraded controller. This implementation does not require any vaults to be redeployed - we can just call setWhitelistedOptionsVault(true) on all existing addresses and it will override the vault type to 2. We already have 4 treasury vaults that are deployed and have funds in them and we would have to redeploy all of them and pass in vault type 2. This would require all of the treasuries to migrate which is undesirable. I agree its a bit hacky.